### PR TITLE
Implement [JENKINS-46151] by adding a system property to support customized metrics sampling interval

### PR DESCRIPTION
**Pull request by svanoort**
Also bump to new parent pom and do FindBugs fixes. 

 [JENKINS-46151](https://issues.jenkins-ci.org/browse/JENKINS-46151)

@reviewbybees

### CODE REVIEW

**Comment by svanoort** [comment]
@stephenc A quickie review when you get back

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: pom.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>1.520</version>
+    <version>2.29</version>
```
Comment (oleg-nenashev) [inline pom.xml:32]: Nobody want to resolve upper bounds fun ;)

**Review by oleg-nenashev — APPROVED** [review]
🐝

**Review by stephenc — APPROVED** [review]

**Comment by svanoort** [comment]
@stephenc Is it ok if I merge?

**Pull request merged**